### ISSUE TITLE
Please add support to download DRM Videos 

### ISSUE THREAD

**Issue by futuregate2016** [heliomarpm/udemy-downloader-gui#196]
### Description of the new feature / enhancement

Sir Please Add option to Download & Decrypt DRM Videos (User must have keys , Like Puyodead1 Tool ) if can just add mp4decrypt & a way to download MPD via yt-dlp or aira2 

your Downloader just missing this 

### Scenario when this would be used?

This will be used when a user want to download DRM Course and he have the keys.

**Comment by juliotec** [comment]
Yes please, I have the client_id.bin (device_client_id_blob) and private_key.pem (device_private_key) files to decrypt the videos, I think its better than ge the decrypted key and decrypted key id

**Comment by heliomarpm** [comment]
> Yes please, I have the client_id.bin (device_client_id_blob) and private_key.pem (device_private_key) files to decrypt the videos, I think its better and faster than get the decrypted key and decrypted key id
> 
> they do the same for crunchy roll drm protected videos
> https://github.com/Crunchy-DL/Crunchy-Downloader
> 
> ![image](https://github.com/user-attachments/assets/ab976c17-94e4-4414-97dd-c882eb1e4ecb)
> 
> 

How does it work?  
Do you only need these 2 files to decrypt any course?

**Comment by babinjemmaobaj3o7u0f5ve** [comment]
> Yes please, I have the client_id.bin (device_client_id_blob) and private_key.pem (device_private_key) files to decrypt the videos, I think its better and faster than get the decrypted key and decrypted key id
> 
> they do the same for crunchy roll drm protected videos https://github.com/Crunchy-DL/Crunchy-Downloader
> 
> ![image](https://private-user-images.githubusercontent.com/7734345/348500528-ab976c17-94e4-4414-97dd-c882eb1e4ecb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIwMTgwMDEsIm5iZiI6MTcyMjAxNzcwMSwicGF0aCI6Ii83NzM0MzQ1LzM0ODUwMDUyOC1hYjk3NmMxNy05NGU0LTQ0MTQtOTdkZC1jODgyZWIxZTRlY2IucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MjZUMTgxNTAxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTA3ZmQ1NTk1ZTIwNzFjM2I1ZGIyZDliNDZkM2U4ZGJlZDFmY2E0OWY2N2MwN2JkYjIwZDAxMDA3ZDA3OTg1YSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.VoNIWHtB4HlmQIuPfSL9DMuvA9xmG_PpYEWi5wLthkU)

i to get those file would you help me. i need a course to download very badly. i will gift some course you also

**Comment by Rutvikballar** [comment]
Yes please add DRM Support. So that people who have keys can Download courses easily

**Comment by kitku2** [comment]
yeah, would be useful

**Comment by brokenman1** [comment]
How can I get the keys, and where should I get them from?

**Comment by Rutvikballar** [comment]
> How can I get the keys, and where should I get them from?
i can't tell you bro...how to get  but  if you want just keys then i will give ...

**Comment by MaikPeters1511** [comment]
Wie bekomme ich den Schlüssel? Alle meine Kurse sind verschlüsselt, die ich gekauft habe. Kann mir jemand helfen? Wie ich die herunterladen kann?

**Comment by Rutvikballar** [comment]
> Wie bekomme ich den Schlüssel? Alle meine Kurse sind verschlüsselt, die ich gekauft habe. Kann mir jemand helfen? Wie ich die herunterladen kann?

ohh, Bruder, ich helfe dir, die Schlüssel für deinen Kurs zu besorgen, wenn du sie wirklich so dringend brauchst.

**Comment by MaikPeters1511** [comment]
lieben gerne...melde dich einfach

**Comment by Rutvikballar** [comment]
> lieben gerne...melde dich einfach

Bruder, das ist meine Telegramm-ID @itzz_RB.

**Comment by MaikPeters1511** [comment]
hab dich angeschrieben

**Comment by MesterPerfect** [comment]
I extracted the key from the course on Udemy. However, the program couldn't download anything.
